backend/module/chaos/serverexperimentation/rtds/rtds_cache_test.go

```diff
@@ -8,74 +8,92 @@ import (
-						},
+func createAbortExperiment(t *testing.T, upstreamCluster string, downstreamCluster string, faultPercent float32, httpStatus int32) *experimentation.Experiment {
+	testConfig := &serverexperimentation.TestSpecification{
+		Config: &serverexperimentation.TestSpecification_Abort{
+			Abort: &serverexperimentation.AbortFault{
+				Target: &serverexperimentation.AbortFault_ClusterPair{
+					ClusterPair: &serverexperimentation.ClusterPairTarget{
+						DownstreamCluster: downstreamCluster,
+						UpstreamCluster:   upstreamCluster,
 					},
-					Percent:    faultPercent,
-					HttpStatus: httpStatus},
-			},
+				},
+				Percent:    faultPercent,
+				HttpStatus: httpStatus},
 		},
 	}
+
+	anyConfig, err := ptypes.MarshalAny(testConfig)
+	if err != nil {
+		t.Errorf("setSnapshot failed %v", err)
+	}
+
+	return &experimentation.Experiment{TestConfig: anyConfig}
 }
 
-func createLatencyExperiment(upstreamCluster string, downstreamCluster string, latencyPercent float32, duration int32) *experimentation.Experiment {
-	return &experimentation.Experiment{
-		TestSpecification: &experimentation.TestSpecification{
-			Config: &experimentation.TestSpecification_Latency{
-				Latency: &experimentation.LatencyFault{
-					Target: &experimentation.LatencyFault_ClusterPair{
-						ClusterPair: &experimentation.ClusterPairTarget{
-							DownstreamCluster: downstreamCluster,
-							UpstreamCluster:   upstreamCluster,
-						},
+func createLatencyExperiment(t *testing.T, upstreamCluster string, downstreamCluster string, latencyPercent float32, duration int32) *experimentation.Experiment {
+	testConfig := &serverexperimentation.TestSpecification{
+		Config: &serverexperimentation.TestSpecification_Latency{
+			Latency: &serverexperimentation.LatencyFault{
+				Target: &serverexperimentation.LatencyFault_ClusterPair{
+					ClusterPair: &serverexperimentation.ClusterPairTarget{
+						DownstreamCluster: downstreamCluster,
+						UpstreamCluster:   upstreamCluster,
 					},
-					Percent:    latencyPercent,
-					DurationMs: duration},
-			},
+				},
+				Percent:    latencyPercent,
+				DurationMs: duration},
 		},
 	}
+
+	anyConfig, err := ptypes.MarshalAny(testConfig)
+	if err != nil {
+		t.Errorf("setSnapshot failed %v", err)
+	}
+
+	return &experimentation.Experiment{TestConfig: anyConfig}
 }
 
-func mockGenerateFaultData() []*experimentation.Experiment {
+func mockGenerateFaultData(t *testing.T) []*experimentation.Experiment {
 	return []*experimentation.Experiment{
-		createAbortExperiment("serviceA", "serviceB", 10, 404),
-		createAbortExperiment("serviceC", "", 20, 504),
-		createLatencyExperiment("serviceA", "serviceD", 30, 100),
-		createLatencyExperiment("serviceE", "", 40, 200),
+		createAbortExperiment(t, "serviceA", "serviceB", 10, 404),
+		createAbortExperiment(t, "serviceC", "", 20, 504),
+		createLatencyExperiment(t, "serviceA", "serviceD", 30, 100),
+		createLatencyExperiment(t, "serviceE", "", 40, 200),
```

Comment:
Do we need to pass `t`? It doesn't look like it's used by `createAbortExperiment()`/`createLatencyExperiment()`

Reply:
it is used by both of these methods for logging errors that can come out of `MarshalAny` method